dspace-uclouvain/src/main/java/org/dspace/uclouvain/authority/UCLouvainPromoterAuthority.java

```diff
@@ -0,0 +1,82 @@
+package org.dspace.uclouvain.authority;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dspace.content.authority.Choice;
+import org.dspace.content.authority.ChoiceAuthority;
+import org.dspace.content.authority.Choices;
+import org.dspace.uclouvain.authority.client.UCLouvainAuthorityClient;
+import org.dspace.uclouvain.authority.configuration.UCLouvainAuthorAuthorityAPIConfiguration;
+import org.dspace.uclouvain.authority.factory.UCLouvainServiceFactory;
+import org.dspace.uclouvain.external.dilbert.model.DialPerson;
+
+public class UCLouvainPromoterAuthority implements ChoiceAuthority {
```

Comment:
This class is very close to UCLouvainAuthorAuthority ; many function are the same.
You could extract all identical code into an Abstract superClass and create children classes based on it.